components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4157,6 +4162,26 @@ public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2
         }
     }
 
+    public static String getIdTokenIssuer(String tenantDomain, String clientId) throws IdentityOAuth2Exception {
+
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                ServiceURLBuilder serviceURLBuilder = ServiceURLBuilder.create();
+                if (SAAS_APP_IDS.contains(clientId)) {
+                    serviceURLBuilder.setSkipCustomDomain(true);
+                }
+
+                return serviceURLBuilder.addPath(OAUTH2_TOKEN_EP_URL).build().getAbsolutePublicURL();
```

Comment:
Alternative way to add the logic.

 return ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL). setSkipCustomDomain(SAAS_APP_IDS.contains(clientId)).build().getAbsolutePublicURL();

Reply:
Updated with [5fc4333](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2458/commits/5fc43339d35766c8996f2ff612eeda9ea75f37e7)